Branch-SDK/src/io/branch/indexing/BranchUniversalObject.java

```diff
@@ -80,6 +105,8 @@ public BranchUniversalObject() {
         type_ = "";
         indexMode_ = CONTENT_INDEX_MODE.PUBLIC; // Default content indexing mode is public
         expirationInMilliSec_ = 0L;
+        price_ = 0.0;
+        currency_ = CURRENCY_TYPE.NONE;
```

Comment:
The reason for initializing the above variables is that this class is implementing parcellable interface. "Null" is not auto unboxed through parcel and it can throw NPE in case of null (Unless we write more code in parcel logic  to handle null with some other default value).